### DISCUSSION TITLE
in Full File View: show file extensions directly after filename

### DISCUSSION THREAD

**Discussion by Pietjw** [doublecmd/doublecmd#1460] [Suggestions]
Full file view: When the file extension is displayed directly after the filename, the Ext column is not needed anymore and the DC window needs less space. Also it improves readability. Similar to Brief File view.

**Comment by j2969719** [comment]
options>file views>columns>custom columns>default>name>field contents>[DC().GETFILENAME{}]

**Reply by Pietjw** [reply]
This works! Thanks.
The Brief file view has for "show file extension" option "directly after filename". You would expect similar for Full file view

**Comment by tomas2t** [comment]
I also want this, but the way Total Commander does it:
When enabling the "show extension directly after filename" I want the Ext column (header) still visible but with no (visible) content, meaning that the Ext column still can be used for sorting. Is this possible? 
Like having an "Invisible" attribute for columns... or why not one setting for "Field contents" (that I could empty in this case) and separate one used for sorting.

**Reply by j2969719** [reply]
you can override the text color for a column, but why bother? even w/o a column, you can always sort by extension using ctrl+f4 or show>sort by ext from the main menu.

**Reply by tomas2t** [reply]
Thanks for the hint! Missed that I can customize colors per column. I'm happy with it as a workaround. Only thing is that the separate extension is visible if the row is both active and selected, but it is fully possible that I've missed some magic combination in color settings.
<img width="352" height="110" alt="image" src="https://github.com/user-attachments/assets/11ee090d-886c-4e07-ad96-1ac959c5d77a" />


Also, thank you for the Ctrl+F4. Just had to change some KDE global shortcuts. With that said, as much as I like and use keyboard shortcuts I want the ext column available when in "mouse mode". Menu fiddling is no option for me for such things.

(Even after 25+ years on Linux, Total Commander is still my biggest (and almost only) missing thing from Windows. Double Commander, that I for some reason haven't tried out until today, is so far the first non-console "commander" that has not made me mad for some reason or another. )

**Reply by j2969719** [reply]
forgot that this wouldnt work with solid cursor. i just prefer frame cursor, and its not noticeable with it. bummer

**Reply by JonnyTech** [reply]
> Also, thank you for the Ctrl+F4. Just had to change some KDE global shortcuts. With that said, as much as I like and use keyboard shortcuts I want the ext column available when in "mouse mode". Menu fiddling is no option for me for such things.

Create a toolbar button and assign it to internal command `cm_sortbyext` - you can then click the other columns to sort their respective contents and click the toolbar button for sorting by extension.

**Reply by tomas2t** [reply]
Yeah, I considered that but I just like it nearby in the header, and that way I don't have to think about or change which panel is active. Also makes the current sorting is clearly visible (except for the icon being black on gray - might look into that some day).

Maybe I'll try out frame cursor, but I think I will just live with the current glitch...

Thx!
